--- libavformat/hls.orig.c	2023-01-13 17:42:19.490267300 +0900
+++ libavformat/hls.c	2023-01-13 17:52:12.165236700 +0900
@@ -1472,6 +1472,7 @@ static int read_data(void *opaque, uint8
     int ret;
     int just_opened = 0;
     int reload_count = 0;
+    int segment_retries = 0;
     struct segment *seg;
 
 restart:
@@ -1563,9 +1564,14 @@ reload:
             av_log(v->parent, AV_LOG_WARNING, "Failed to open segment %"PRId64" of playlist %d\n",
                    v->cur_seq_no,
                    v->index);
-            v->cur_seq_no += 1;
+            if (segment_retries++ >= 2)
+                return AVERROR_EXIT;
+            av_log(v->parent, AV_LOG_WARNING, "Retry to open segment %"PRId64" of playlist %d\n",
+                   v->cur_seq_no,
+                   v->index);    
             goto reload;
         }
+        segment_retries = 0;
         just_opened = 1;
     }
 
